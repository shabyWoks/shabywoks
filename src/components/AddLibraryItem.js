import React from 'react';
import {connect} from 'react-redux';
import LibraryItemForm from './LibraryItemForm';
import {addLibraryItem} from '../actions/libraryA';

export class AddLibraryItem extends React.Component {
    onSubmit = (libraryItem) => {
        this.props.addLibraryItem(libraryItem);
        this.props.history.push('/');
    }
    render () {
        return (
            <div>
                <LibraryItemForm
                    onSubmit={this.onSubmit} 
                />
            </div>
        );
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        addLibraryItem: (libraryItem) => dispatch(addLibraryItem(libraryItem))
    }
}

const ConnectedAddLibraryItem = connect(undefined, mapDispatchToProps)(AddLibraryItem);
export default ConnectedAddLibraryItem;