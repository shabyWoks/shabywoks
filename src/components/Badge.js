import React from 'react';

const Badge = (props) => {
  return (
    <div className="qwb-body">
      <img src={props.src} />
      <span>{props.tagName}</span>
    </div>
  )
}

export default Badge;