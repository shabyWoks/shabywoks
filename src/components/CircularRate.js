import React from 'react';

class CircularRate extends React.Component {
    getCircle = (num) => {
        let value = 'polygon(50% 50%, 50% 0%, ';
        if(num <= 12.5){
            let v = (4 * num) + 50;
            value = value + `${v}% 0%`;
        } else if(num <= 37.5){
            value = value + '100% 0%, ';
            let v = (4 * (num - 12.5));
            value = value + `100% ${v}%`;
        } else if(num <= 62.5){
            value = value + '100% 0%, 100% 100%, ';
            let v = (4 * (num - 37.5));
            value = value + `${100 - v}% 100%`;
        } else if(num <= 87.5){
            value = value + '100% 0%, 100% 100%, 0% 100%, ';
            let v = (4 * (num - 62.5));
            value = value + `0% ${100 - v}%`;
        } else {
            value = value + '100% 0%, 100% 100%, 0% 100%, 0% 0%, ';
            let v = (4 * (num - 87.5));
            value = value + `${v}% 0%`;
        }
        value = value + ')';
        return value;
    }
    render () 
    {
        return (
            <div className="flex-start flex-down">
                <div 
                    className="flex-center flex-down rate-meter"
                    style= {
                        { 
                            clipPath: this.getCircle(this.props.value),
                            backgroundColor: `${this.props.color}22`, 
                            border: `0.7rem solid ${this.props.color}88` 
                        }
                    }
                >
                </div>
                <div className="flex-center rate-meter rate-meter-value" style={{color: `${this.props.color}44`}}>
                    {`${this.props.value}%`}
                </div>
                <div>
                   <p className="p-rateline-1">{this.props.name.toUpperCase()}</p>
                </div>
            </div>
        )
    }
}
export default CircularRate;