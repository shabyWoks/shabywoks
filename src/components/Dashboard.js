import React from 'react';

class Dashboard extends React.Component {
    selected= 1;
    start = [45, 135, 225, 315];
    constructor(props){
        super(props);
        this.state = {
            start: [45, 135, 225, 315]
        }
    }

    addNum = (num) => {
        for(let i =0; i< this.start.length; i++){
            this.start[i] = this.start[i] + num;
        }
        
    }
    rotate = (val) => {
        if(val > this.selected){
            this.selected === 1 && val === 4 ? this.addNum(90) : this.addNum(-90);
        } else if(val < this.selected){
            this.selected === 4 && val === 1 ? this.addNum(-90) : this.addNum(90);
        }
        this.selected = val;
        this.setState(() => ({ start: this.start }));
    }

    renderContent = () => {
        if(this.selected === 1){
            return (
                <div className="pie-detail">
                    <p>Shubham Bhiwaniwala</p>
                    <p>Software Engineer at YOI.</p>      
                </div>
            );
        } else if(this.selected === 2){
            return (
                <div className="pie-detail">
                    <p>Bachelor in Engineering</p>
                    <p>Jadavpur University, India.</p>      
                </div>
            );
        } else if(this.selected === 3){
            return (
                <div className="pie-detail">
                    <p>Intern</p>
                    <p>Goldman Sachs.</p>      
                </div>
            );
        } else if(this.selected === 4){
            return (
                <div className="pie-detail">
                    <p>Dedicated</p>
                    <p>Enthusiast</p>
                    <p>Coder,</p>
                    <p>Developer, and</p>
                    <p>Learner.</p>      
                </div>
            );
        }
    }

    render() {
        return (
            <div className='dashboard-class'>
                <div className="parallax">
                    <div className="pie-component">
                        <div className= {`pie`} style= {{transform : `rotate(${this.start[0]}deg)`}} onClick={this.rotate.bind(this, 1)}>
                            <div className= "pie-content">Introducing</div>
                        </div>
                        <div className= {`pie`} style= {{transform : `rotate(${this.start[1]}deg)`}} onClick={this.rotate.bind(this, 2)}>
                            <div className= "pie-content">Education</div>
                        </div>
                        <div className= {`pie`} style= {{transform : `rotate(${this.start[2]}deg)`}} onClick={this.rotate.bind(this, 3)}>
                            <div className= "pie-content">Experience</div>
                        </div>
                        <div className= {`pie`} style= {{transform : `rotate(${this.start[3]}deg)`}} onClick={this.rotate.bind(this, 4)}>
                            <div className= "pie-content">That's Me</div>
                        </div>
                    </div>
                    { this.renderContent() }
                </div>
            </div>
        );
    }
    
}

export default Dashboard;