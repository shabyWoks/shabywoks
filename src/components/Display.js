import React from 'react';
import Question from '../components/question/Question';

class Display extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      left  : 0,
      right : 3,
      count : props.item.items.length
    }
  }
  startPos = 0;

  displayQuestions = () => {
    return this.props.item.items.map((question) => {
      return <Question questionDetails= {question} badge={this.badge} user={this.user} tags={this.tags} baseImageUrl = {this.baseImageUrl}/>
    });
  }

  onLeftMoveClick = () => {
    let uiWidth = document.getElementsByClassName('display-holder')[0].clientWidth;
    let count   = 4;
    if(this.state.left <= 0) {
      return;
    }
    if (this.state.left - 4 < 0) {
      count = this.state.left
    }
    uiWidth       = (count/4) * uiWidth;
    this.startPos = this.startPos + uiWidth;
    this.setState((state) => {
      return {
        ...state,
        left  : state.left  - count,
        right : state.right - count
      }
    })
    this.translateXY(document.getElementById('display-holder-main'), this.startPos);
  }

  translateXY (elem, xPos) {
    elem.style.transition = `500ms`;
    elem.style.transform = `translateX(${xPos}px)`;
  }

  onRightMoveClick = () => {
    let uiWidth = document.getElementsByClassName('display-holder')[0].clientWidth;
    let count   = 4;
    if(this.state.right >= this.state.count - 1) {
      return;
    }
    if (this.state.right + 4 >= this.state.count - 1) {
      count = this.state.count - this.state.right - 1
    }
    uiWidth       = (count/4)*uiWidth;
    this.startPos = this.startPos - uiWidth;
    this.setState((state) => {
      return {
        ...state,
        left  : state.left  + count,
        right : state.right + count
      }
    })
    this.translateXY(document.getElementById('display-holder-main'), this.startPos);
  }

  render () {
    return (
      <div className="display-holder">
        <span className={`arrow left-arrow ${this.state.left == 0 ? "hide" : "" }`} onClick= {this.onLeftMoveClick}>
          <img src="/images/leftArrow.png" width="30px" height="30px" style= {{ padding: '10px', paddingLeft: '7px' }} />
        </span>
        <span className={`arrow right-arrow ${this.state.right == this.state.count - 1 ? "hide" : ""} `} onClick= {this.onRightMoveClick}>
        <img src="/images/rightArrow.png" width="30px" height="30px" style= {{ padding: '10px', paddingRight: '7px' }} />
        </span>
        <div>{this.props.item.name}</div>
        <div style={{overflow:'hidden', height: '100%'}}>
          <div id="display-holder-main" style={{height: '100%', whiteSpace: 'nowrap'}}>
          {
            this.displayQuestions()
          }
          </div>
        </div>
      </div>
    )
  }
}

export default Display;