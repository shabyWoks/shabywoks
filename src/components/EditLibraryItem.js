import React from 'react';
import {connect} from 'react-redux';
import LibraryItemForm from './LibraryItemForm';
import { editLibraryItem, rempveLibraryItem } from '../actions/libraryA';

class EditLibraryItem extends React.Component {
    onSubmit = (libraryItem) => {
        this.props.dispatch(editLibraryItem(libraryItem));
        this.props.history.push("/");
    }
    onClick = () => {
        this.props.dispatch(rempveLibraryItem(this.props.libraryItem.id));
        this.props.history.push("/");
    }
    render () {
        return (
            <div>
                <LibraryItemForm onSubmit={this.onSubmit} libraryItem={this.props.libraryItem} />
                <button onClick={this.onClick}>Remove</button>
            </div>
        );
    }
}

const mapStateToProps = (state, props) => {
    return {
        libraryItem: state.library.libraryItems.find((libraryItem) => libraryItem.id === props.match.params.id)
    }
}

const ConnectedEditLibraryItem = connect(mapStateToProps)(EditLibraryItem)
export default ConnectedEditLibraryItem;