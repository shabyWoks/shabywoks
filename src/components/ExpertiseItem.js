import React from 'react';



const ExpertiseItem = (props) => {
    return (
        <div className="flex-center flex-down" style={{marginBottom: '3.2rem'}}>
            <div className="expertise-item-body">
                <img src= {props.imageUrl}/>
            </div>
            <div className="expertise-rating">
                <div className="meter" style={{backgroundColor: `${props.color}88`, width: `${props.accuracy}%`}}></div>
            </div>
        </div>
    )
}

export default ExpertiseItem;