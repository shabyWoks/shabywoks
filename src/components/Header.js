import React from 'react';
import {Link} from 'react-router-dom';

const Header = (props) => {
    return (
        <div>
            <ul>
                <Link to="/" > Dashboard </Link>
                <Link to="/my-library" > My Library </Link>
            </ul>
        </div>
    )
}

export default Header;