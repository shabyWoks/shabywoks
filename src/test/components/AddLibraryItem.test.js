import React from 'react';
import {shallow} from 'enzyme';
import {AddLibraryItem} from '../../components/AddLibraryItem';
import libraryItems from '../fixtures/libraryItems';

let addLibraryItem, history, wrapper;

beforeEach(() => {
    addLibraryItem = jest.fn();
    history = {push: jest.fn()};
    wrapper = shallow(<AddLibraryItem addLibraryItem={addLibraryItem} history={history}/>);
    
})

test('should render add library item correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should execute on submit button successfully', () => {
    wrapper.find('LibraryItemForm').prop('onSubmit')(libraryItems[0]);
    expect(addLibraryItem).toHaveBeenLastCalledWith(libraryItems[0]);
    expect(history.push).toHaveBeenLastCalledWith("/");
});