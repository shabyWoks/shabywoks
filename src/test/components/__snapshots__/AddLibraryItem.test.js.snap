// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render add library item correctly 1`] = `
<div>
  <Connect(LibraryItemForm)
    onSubmit={[Function]}
  />
</div>
`;
